glusterd2/commands/volumes/volume-smartvol-txn.go

```diff
@@ -48,15 +49,33 @@ func txnPrepareBricks(c transaction.TxnCtx) error {
 			}
 
 			// LV Creation
-			err = deviceutils.CreateLV(b.VgName, b.TpName, b.LvName, b.Size)
-			if err != nil {
-				c.Logger().WithError(err).WithFields(log.Fields{
-					"vg-name": b.VgName,
-					"tp-name": b.TpName,
-					"lv-name": b.LvName,
-					"size":    b.Size,
-				}).Error("lvcreate failed")
-				return err
+			if b.Type == "arbiter" {
+				arbiterSize, err := bricksplanner.ArbiterBrickSizeCalculation(b.Size, req.AverageFileSize)
+				if err != nil {
+					c.Logger().WithError(err).Error("average file size is zero")
+					return err
+				}
+				err = deviceutils.CreateLV(b.VgName, b.TpName, b.LvName, arbiterSize)
+				if err != nil {
+					c.Logger().WithError(err).WithFields(log.Fields{
+						"vg-name": b.VgName,
+						"tp-name": b.TpName,
+						"lv-name": b.LvName,
+						"size":    arbiterSize,
+					}).Error("lvcreate failed")
+					return err
+				}
+			} else {
+				err = deviceutils.CreateLV(b.VgName, b.TpName, b.LvName, b.Size)
+				if err != nil {
+					c.Logger().WithError(err).WithFields(log.Fields{
+						"vg-name": b.VgName,
+						"tp-name": b.TpName,
+						"lv-name": b.LvName,
+						"size":    b.Size,
+					}).Error("lvcreate failed")
+					return err
+				}
```

Comment:
Add mkfs option if b.Type is arbiter where `deviceutils.MakeXfs` is called, as @JohnStrunk suggested